smart_simulation/database/db_utils.py

```diff
@@ -0,0 +1,84 @@
+import logging
+import pathlib
+import sqlite3
+
+import pandas as pd
+
+import daiquiri
+from smart_simulation.cfg_templates.config import package_dir
+
+daiquiri.setup(
+    level=logging.INFO,
+    outputs=(
+        daiquiri.output.Stream(
+            formatter=daiquiri.formatter.ColorFormatter(
+                fmt="%(asctime)s [%(levelname)s] %(name)s.%(" "funcName)s: %(message)s"
+            )
+        ),
+    ),
+)
+
+PACKAGE_PATH = pathlib.Path(package_dir)
+LOCAL_PROD_DB = PACKAGE_PATH / "smart_simulation" / "database" / "production.db"
+
+
+def connect(database: pathlib.PurePath = LOCAL_PROD_DB) -> sqlite3.connect:
```

Comment:
Database connections are a good opportunity for memoization / caching. Python 3.6 has lru_cahce and 3.7 has `cache` which you can decorate the function with https://docs.python.org/3.6/library/functools.html#functools.lru_cache

Reply:
Also, I think the sqlite3 connection can provide caching itself (through the same mechanism I'm guessing).